File: docs/aes_helper.py
import base64
from cryptography.hazmat.primitives.ciphers import Cipher, algorithms, modes
from cryptography.hazmat.primitives.padding import PKCS7


class AesHelper:
    """AES/CBC/PKCS7 加解密工具"""

    # AES-192 (Key 24字节)
    _KEY = "k9Xm2pQ7wR4vB8nL3yT6jF0h".encode("utf-8")
    # IV 16字节
    _IV = "aZ3cE5gH8jK2mN4p".encode("utf-8")

    @classmethod
    def encrypt(cls, plain_text: str) -> str:
        """
        加密：明文 → Base64密文

        Args:
            plain_text: 明文

        Returns:
            Base64编码的密文
        """
        plain_bytes = plain_text.encode("utf-8")

        # PKCS7 填充
        padder = PKCS7(algorithms.AES.block_size).padder()
        padded_data = padder.update(plain_bytes) + padder.finalize()

        cipher = Cipher(algorithms.AES(cls._KEY), modes.CBC(cls._IV))
        encryptor = cipher.encryptor()
        cipher_bytes = encryptor.update(padded_data) + encryptor.finalize()

        return base64.b64encode(cipher_bytes).decode("utf-8")

    @classmethod
    def decrypt(cls, cipher_text: str) -> str:
        """
        解密：Base64密文 → 明文

        Args:
            cipher_text: Base64编码的密文

        Returns:
            明文
        """
        cipher_bytes = base64.b64decode(cipher_text)

        cipher = Cipher(algorithms.AES(cls._KEY), modes.CBC(cls._IV))
        decryptor = cipher.decryptor()
        padded_data = decryptor.update(cipher_bytes) + decryptor.finalize()

        # 去除 PKCS7 填充
        unpadder = PKCS7(algorithms.AES.block_size).unpadder()
        plain_bytes = unpadder.update(padded_data) + unpadder.finalize()

        return plain_bytes.decode("utf-8")


# ===== 使用示例 =====
if __name__ == "__main__":
    original = "15222506078"

    # 加密
    encrypted = AesHelper.encrypt(original)
    print(f"密文: {encrypted}")

    # 解密
    decrypted = AesHelper.decrypt(encrypted)
    print(f"明文: {decrypted}")

    # 验证
    print("✓ 加解密一致" if original == decrypted else "✗ 不匹配")

File: docs/AesHelper.cs
using System;
using System.IO;
using System.Security.Cryptography;
using System.Text;

/// <summary>
/// AES/CBC/PKCS7 加解密工具
/// </summary>
public static class AesHelper
{
    // AES-192 (Key 24字节)
    private static readonly byte[] Key = Encoding.UTF8.GetBytes("k9Xm2pQ7wR4vB8nL3yT6jF0h");
    // IV 16字节
    private static readonly byte[] IV = Encoding.UTF8.GetBytes("aZ3cE5gH8jK2mN4p");

    /// <summary>
    /// 加密：明文 → Base64密文
    /// </summary>
    /// <param name="plainText">明文</param>
    /// <returns>Base64编码的密文</returns>
    public static string Encrypt(string plainText)
    {
        byte[] plainBytes = Encoding.UTF8.GetBytes(plainText);
        using (Aes aes = Aes.Create())
        {
            aes.Key = Key;
            aes.IV = IV;
            aes.Mode = CipherMode.CBC;
            aes.Padding = PaddingMode.PKCS7;
            using (var encryptor = aes.CreateEncryptor())
            using (var ms = new MemoryStream())
            {
                using (var cs = new CryptoStream(ms, encryptor, CryptoStreamMode.Write))
                {
                    cs.Write(plainBytes, 0, plainBytes.Length);
                    cs.FlushFinalBlock();
                }
                return Convert.ToBase64String(ms.ToArray());
            }
        }
    }

    /// <summary>
    /// 解密：Base64密文 → 明文
    /// </summary>
    /// <param name="cipherText">Base64编码的密文</param>
    /// <returns>明文</returns>
    public static string Decrypt(string cipherText)
    {
        byte[] cipherBytes = Convert.FromBase64String(cipherText);
        using (Aes aes = Aes.Create())
        {
            aes.Key = Key;
            aes.IV = IV;
            aes.Mode = CipherMode.CBC;
            aes.Padding = PaddingMode.PKCS7;
            using (var decryptor = aes.CreateDecryptor())
            using (var ms = new MemoryStream(cipherBytes))
            using (var cs = new CryptoStream(ms, decryptor, CryptoStreamMode.Read))
            using (var sr = new StreamReader(cs, Encoding.UTF8))
            {
                return sr.ReadToEnd();
            }
        }
    }
}
